File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n = s.size();
        int result = 0;
        for(int i=0; i<n; i++){
            int left = i;
            int right = i;

            while(left >= 0 && right < n && s[left] == s[right]){
                result++;
                left--;
                right++;
            } 

            left = i-1;
            right = i;
            while(left >= 0 && right < n && s[left] == s[right]){
                result++;
                left--;
                right++;
            } 
        }
        return result;
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        """
            0   1   2   3
            |           |
                |
            1   2   4   8
            10  11  12  13
            14  20  30  40

            find 10
        """
        
        left = 0
        right = len(matrix[0])-1
        row = 0

        while row < len(matrix) and (left <= right):
            mid = left + (right-left)//2
            key = matrix[row][mid]
            rightKey = matrix[row][right]

            if target > rightKey:
                row = row + 1

            elif key == target:
                return True
            
            elif target > key:
                left = mid+1
            
            else:
                right = mid-1

        return False